islandora_transforms/library/qualified_date_range.xslt

```diff
@@ -72,18 +95,56 @@
       <!-- If we managed to get a date range, make some fields. Single-valued
            dates count as a 'range' of one year. -->
       <xsl:if test="$low_end != '' and $high_end != ''">
-        <xsl:call-template name="general_mods_field">
-          <xsl:with-param name="prefix" select="concat($prefix, local-name(), '_qualified_range_low_end_')"/>
-          <xsl:with-param name="suffix" select="$suffix"/>
-          <xsl:with-param name="value" select="$low_end"/>
-        </xsl:call-template>
-        <xsl:call-template name="general_mods_field">
-          <xsl:with-param name="prefix" select="concat($prefix, local-name(), '_qualified_range_high_end_')"/>
-          <xsl:with-param name="suffix" select="$suffix"/>
-          <xsl:with-param name="value" select="$high_end"/>
+        <xsl:call-template name="qdr_year_loop">
+          <xsl:with-param name="prefix" select="concat($prefix, local-name(), '_')"/>
+          <xsl:with-param name="start" select="$low_end"/>
+          <xsl:with-param name="end" select="$high_end"/>
         </xsl:call-template>
       </xsl:if>
     </xsl:if>
   </xsl:template>
 
+  <!-- Year-looping template. -->
+  <!-- XXX: As selecting "x to y" isn't implemented until XSLT 2.0, and there's
```

Comment:
This isn't the most clear comment: perhaps something like "Since our XSLT version doesn't iterate over ranges and our XSLT processor doesn't support tail recursion recognition we are using divide and conquer recursion to lower the call stack depth."